pyresample/geometry.py

```diff
@@ -1887,7 +1889,8 @@ def projection_y_coords(self):
     def outer_boundary_corners(self):
         """Return the lon,lat of the outer edges of the corner points."""
         from pyresample.spherical_geometry import Coordinate
-        proj = Proj(**self.proj_dict)
+        proj_def = self.crs_wkt if hasattr(self, 'crs_wkt') else self.proj_dict
```

Comment:
Should we have a shortcut method for this? This seems to be used in multiple places

Reply:
I think it would be better long term to just say from now on pyresample requires pyproj 2.2+ (I think that's what we need for all of the CRS stuff to work). This would drop the possibility of not having a `.crs` attribute. Maybe not in this PR though? Maybe we do one pyresample release with this resampler (and some of the other existing PRs) and then force pyproj versions?

Reply:
If you don't want to force the pyproj version in this PR, could you then factorize this line and it's duplicates?

Reply:
Ok I refactored this into two properties: one for WKT/string and one for CRS/dict. I also added an `is_geostationary` property to workaround one of the required uses of `proj_dict`.